src/verify/babe_current_epoch.rs

```diff
@@ -0,0 +1,168 @@
+// Substrate-lite
+// Copyright (C) 2019-2021  Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::{
+    chain::chain_information::BabeEpochInformation,
+    executor::{host, read_only_runtime_host},
+    header,
+};
+
+use alloc::vec::Vec;
+use parity_scale_codec::{Decode, Encode};
+
+pub struct Config {
+    /// Runtime used to get the current babe epoch. Must be built using the Wasm code found at the
+    /// `:code` key of the parent block storage.
+    pub parent_runtime: host::HostVmPrototype,
+}
+
+#[derive(Debug, Clone, derive_more::Display)]
+pub enum Error {
+    /// Error while starting the Wasm virtual machine.
+    #[display(fmt = "{}", _0)]
+    WasmStart(host::StartErr),
+    /// Error while running the Wasm virtual machine.
+    #[display(fmt = "{}", _0)]
+    WasmVm(read_only_runtime_host::Error),
+    /// Error while decoding the babe epoch.
+    #[display(fmt = "{}", _0)]
+    DecodeFailed(parity_scale_codec::Error),
+}
+
+/// Fetches the current babe epoch using `BabeApi_current_epoch`.
+pub fn babe_current_epoch(config: Config) -> Verify {
+    let vm = read_only_runtime_host::run(read_only_runtime_host::Config {
+        virtual_machine: config.parent_runtime,
+        function_to_call: "BabeApi_current_epoch",
+        // `BabeApi_current_epoch`doesn't take any parameters.
+        parameter: std::iter::empty::<&[u8]>(),
+    });
+
+    match vm {
+        Ok(vm) => Verify::from_inner(vm),
+        Err(err) => Verify::Finished(Err(Error::WasmStart(err))),
+    }
+}
+
+/// Current state.
+#[must_use]
+pub enum Verify {
```

Comment:
```suggestion
pub enum Query {
```

It's not a verification. We're not verifying anything.